contracts/handlers/ERC20Handler.sol

```diff
@@ -86,54 +89,61 @@ contract ERC20Handler is IDepositExecute, HandlerHelpers, ERC20Safe {
         recipientAddress            bytes       bytes  64 - END
         @dev Depending if the corresponding {tokenAddress} for the parsed {resourceID} is
         marked true in {_burnList}, deposited tokens will be burned, if not, they will be locked.
+        @notice Emits {ERC20InitiateBridge} event with recipientAddress, resourceId, and amount.
      */
-    function deposit(
-        bytes32 resourceID,
-        uint8   destinationChainID,
-        uint64  depositNonce,
-        address depositer,
-        bytes   calldata data
-    ) external override onlyBridge {
-        bytes   memory recipientAddress;
-        uint256        amount;
-        uint256        lenRecipientAddress;
-
-        assembly {
-
-            amount := calldataload(0xC4)
-
-            recipientAddress := mload(0x40)
-            lenRecipientAddress := calldataload(0xE4)
-            mstore(0x40, add(0x20, add(recipientAddress, lenRecipientAddress)))
-
-            calldatacopy(
-                recipientAddress, // copy to destinationRecipientAddress
-                0xE4, // copy from calldata @ 0x104
-                sub(calldatasize(), 0xE) // copy size (calldatasize - 0x104)
-            )
-        }
-
-        address tokenAddress = _resourceIDToTokenContractAddress[resourceID];
-        require(_contractWhitelist[tokenAddress], "provided tokenAddress is not whitelisted");
-
-        if (_burnList[tokenAddress]) {
-            burnERC20(tokenAddress, depositer, amount);
-        } else {
-            lockERC20(tokenAddress, depositer, address(this), amount);
-        }
-
-        _depositRecords[destinationChainID][depositNonce] = DepositRecord(
-            tokenAddress,
-            uint8(lenRecipientAddress),
-            destinationChainID,
-            resourceID,
-            recipientAddress,
-            depositer,
-            amount
-        );
+  function deposit(
+    bytes32 resourceID,
+    uint8 destinationChainID,
+    uint64 depositNonce,
+    address depositer,
+    bytes calldata data
+  ) external override onlyBridge {
+    bytes memory recipientAddress;
+    uint256 amount;
+    uint256 lenRecipientAddress;
+
+    assembly {
+      amount := calldataload(0xC4)
+
+      recipientAddress := mload(0x40)
+      lenRecipientAddress := calldataload(0xE4)
+      mstore(0x40, add(0x20, add(recipientAddress, lenRecipientAddress)))
+
+      calldatacopy(
+        recipientAddress, // copy to destinationRecipientAddress
+        0xE4, // copy from calldata @ 0x104
+        sub(calldatasize(), 0xE) // copy size (calldatasize - 0x104)
+      )
     }
 
-    /**
+    address tokenAddress = _resourceIDToTokenContractAddress[resourceID];
+    require(_contractWhitelist[tokenAddress], 'provided tokenAddress is not whitelisted');
+
+    if (_burnList[tokenAddress]) {
+      // effectively 'burn' ERC20 that does not implement ERC20Burnable 'burnFrom()'
+      lockERC20(tokenAddress, depositer, address(0x000000000000000000000000000000000000dEaD), amount);
```

Comment:
Change the burn technique to send to a dead address to support tokens that don't support burnFrom